asf_public_discourse_home_decarbonisation/pipeline/faqs_identification/extract_questions.py

```diff
@@ -0,0 +1,445 @@
+        inclusion_phrases (List[str]): A list of inclusion phrases. Each item in the list should be a string representing a phrase to search for in the "I don't know" phrases.
+
+    Returns:
+        sentences_without_inclusion (List[str]):
+            - This list contains sentences extracted from the "I don't know" phrases, excluding the inclusion phrases.
+
+    Note:
+        - The function is case-insensitive when searching for the inclusion phrases.
+        - If an inclusion phrase is not found in a sentence, that sentence will not be included in the output list.
+    """
+    sentences_without_inclusion = []
+    for sentence in idk_phrases:
+        for phrase in inclusion_phrases:
+            if phrase.lower() in sentence.lower():
+                start_index = sentence.lower().index(phrase.lower()) + len(phrase)
+                extracted_text = sentence[start_index:].strip()
+                sentences_without_inclusion.append(extracted_text)
+                break
+    return sentences_without_inclusion
+
+
+def logger_statistics(category_dataframe):
+    """
+    Calculates and logs basic statistics about the category-specific data.
+
+    Args:
+        category_dataframe (DataFrame): The category-specific data.
+
+    Note:
+        - The function calculates the total number of "idk" phrases, questions, and sentences in the dataframe.
+        - The statistics are logged using the `logging.info()` method.
+
+    Example:
+        logger_statistics(category_dataframe)
+
+    """
+    total_number_of_idk_phrases = category_dataframe["idk_phrases"].apply(len).sum()
+    total_number_of_sentences = category_dataframe["text"].apply(len).sum()
+    total_questions = category_dataframe["questions"].apply(len).sum()
+
+    # Print some basic statistics
+    logging.info(f"Total number of idk phrases: {total_number_of_idk_phrases}")
+    logging.info(f"Total number of questions: {total_questions}")
+    logging.info(f"Total number of sentences: {total_number_of_sentences}")
+
+
+if __name__ == "__main__":
+    # Read the CSV file
+    args = create_argparser()
+    category = args.category
+    collection_date_time = args.collection_date_time
+    num_of_words = args.num_of_words
+    post_type = args.post_type
+    # uses different getter function depending on the forum
+    category_dataframe = get_category_data(args.forum, category, collection_date_time)
+    FORUM_FAQ_PATH = os.path.join(
+        PROJECT_DIR, f"outputs/data/extracted_questions/{args.forum}/forum_{category}/"
+    )
+    # Ensure the output directory exists
+    os.makedirs(FORUM_FAQ_PATH, exist_ok=True)
+    question_words = [
+        "what",
+        "how",
+        "where",
+        "when",
+        "who",
+        "why",
+        "which",
+        "really",
+        "any",
+        "what's",
+        "how's",
+        "really",
+        "thoughts",
+        "is",
```

Comment:
Taking a brief look at the data, most of what we get aren't actually questions!
Most of sentences start with "is" but they are not questions, so we should probably remove at least "is" from this list.
I'll take a longer look at the data using the files from your other PR, but I think a lot of what we're getting as questions isn't useful at all.

Reply:
I've deleted:"really","thoughts","is", !

Reply:
Alright - are results better? Are the sentences coming up from this list as questions, really questions?

Reply:
I've changed the "extracting questions" methodology to the following:
-Sentence must end with "?"
-At least from my perspective, it makes the question words list redundant as we will want to capture every sentence which ends with a "?", unless we want to cut down further. 
-Minimum of 3 words.

Reply:
The output for BH UFH looks like a 'clean' data-set when I scroll the csv.